app/apphandlers/processorAppHandler.go

```diff
@@ -169,14 +169,23 @@ func (a *processorApp) StartRudderCore(ctx context.Context, options *app.Options
 		jobsdb.WithSkipMaintenanceErr(config.GetBool("BatchRouter.jobsDB.skipMaintenanceError", false)),
 	)
 	defer batchRouterDB.Close()
-	errDB := jobsdb.NewForReadWrite(
+	errDBForRead := jobsdb.NewForRead(
 		"proc_error",
 		jobsdb.WithClearDB(options.ClearDB),
 		jobsdb.WithPreBackupHandlers(prebackupHandlers),
 		jobsdb.WithDSLimit(&a.config.processorDSLimit),
 		jobsdb.WithFileUploaderProvider(fileUploaderProvider),
 		jobsdb.WithSkipMaintenanceErr(config.GetBool("Processor.jobsDB.skipMaintenanceError", false)),
 	)
+	defer errDBForRead.Close()
+	errDBForWrite := jobsdb.NewForWrite(
+		"proc_error",
+		jobsdb.WithClearDB(options.ClearDB),
```

Comment:
I think this needs to be included too
```
jobsdb.WithSkipMaintenanceErr(config.GetBool("Processor.jobsDB.skipMaintenanceError", false)),
```

Reply:
Actually, writer doesn't perform any maintenance, so not required right?

Reply:
`dsMigrationLoop` is part of jobsDB's maintenance

Reply:
Sorry, my previous comment was inaccurate. Here is a list of maintenance operations:
- **Writer** jobsdb: `addNewDSLoop`
- **Reader** jobsdb: `backupDSLoop`, `refreshDSListLoop`, `migrateDSLoop`